drivers/hmis/app/models/hmis/ce/referral.rb

```diff
@@ -132,5 +134,16 @@ def ce_template
 
       errors.add(:workflow_instance, 'must be a CE template')
     end
+
+    def consistent_data_source
+      msg = 'must be in same data source as opportunity'
+
+      # Opportunity takes the workflow template data source as the "source of truth", so do the same here
+      data_source = opportunity.workflow_template.data_source
+
+      errors.add(:client, msg) unless data_source == client.data_source
+      errors.add(:target_enrollment, msg) if target_enrollment && data_source != target_enrollment.data_source
+      # Source enrollment doesn't necessarily need to be in the same data source as the opportunity
```

Comment:
To confirm - is this correct, that the source enrollment could be in a different data source?